### ISSUE TITLE
Consider a desktop daemon for notifications

### ISSUE THREAD

**Issue by turt2live** [vector-im/riot-web#11325]
To keep the task tray (or icon on macos) accurate all the time.

**Comment by dbkr** [comment]
Given Riot continues to run 'in the tray' (unless you turn the tray icon off) isn't this just 'start at login' and starting minimised?

**Comment by turt2live** [comment]
possibly, though Riot isn't exactly known for light memory usage. We could work around that by having a daemon that has the sole purpose of handling notifications (being a push target) and the user can open the app when needed. This also makes the close button actually close the app, returning resources to the system.

Alternatively, we rework our event handling (again) to stop using so much memory :|

**Comment by BombX** [comment]
> Given Riot continues to run 'in the tray' (unless you turn the tray icon off) isn't this just 'start at login' and starting minimised?

On macOS the 'start at login' and 'starting minimised' are 2 extra-actions to be put in place by user. I think that average users are not expecting o 'tweak' the behavior of an App to just have an IM App that should be up-to-date and accurate all the time like the App on iOS.

**Comment by BombX** [comment]
> possibly, though Riot isn't exactly known for light memory usage. We could work around that by having a daemon that has the sole purpose of handling notifications (being a push target) and the user can open the app when needed. This also makes the close button actually close the app, returning resources to the system.

For me it is the perfect solution. With a behavior like that, I could convince an average user (like my family) to switch from App like WhatsApp -> Riot. Indeed, the basic thing an average user is expecting is to be informed of message from their contacts, because they want to participate to the conversation. Having to open the App on desktop to be up-to-date with other is very annoying, because the risk is to miss things when the App is closed :-/ And tweaking the behavior by asking to add 'start at login' and then select 'starting minimized' looks like a low level App (sorry) for people that don't know how it really works IMHO (the economy, difficulties and science of programming...)
Continue your great works, next year WhatsApp may have adverts within the conversation, a great time to make family and friends change their App, so don't miss the opportunities and fix things like that to be prepared!

**Comment by turt2live** [comment]
I don't think this is needed anymore - if folks disagree, please open a new issue with more details regarding the request.
